subsys/caf/modules/power_manager.c

```diff
@@ -28,10 +28,19 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_POWER_MANAGER_LOG_LEVEL);
 #include <caf/events/keep_alive_event.h>
 #include <caf/events/force_power_down_event.h>
 
+#if IS_ENABLED(CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET)
```

Comment:
I prefer this is not done.

Let's always include the header and declare these variables with the same access limitation. Will make the code easier to maintain.

Reply:
Maintenance issues are costly. My opinion is that it should be on par with performance.
Sometimes it's even better to pay a bit with performance to get code that is much more better maintained.

Reply:
Alright! Just to check if I understood this: The more maintainable choice to make here, is to just use the lines inside the if-clause, removing the if-else stuff that's there now?

Reply:
In this case global variables were defined in both cases. What preprocessor removed was only including header and defining two macros - both things with only impact compilation time and the impact will it have considering modern cpus will be zero.

On the other hand if you add the preprocessor check somebody reading this code will have to branch mentally. Will have to check both tracks to keep everything understood. This is especially problematic - people don't deal well with such things and code that branches a lot quickly becomes very bug prone. https://en.wikipedia.org/wiki/Cyclomatic_complexity

Additionally when code is built it will also have to be built for both cases to ensure both branches are error free.